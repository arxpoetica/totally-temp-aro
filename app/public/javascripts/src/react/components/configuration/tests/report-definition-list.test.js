/* global test expect jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ReportDefinitionList } from '../report-definition-list'

const reportsState = {
  metaData: [
    { 'id': 1, 'reportType': 'GENERAL', 'name': 'plan_report', 'displayName': 'plan_report', 'media_types': ['kml'] },
    { 'id': 2, 'reportType': 'GENERAL', 'name': 'planned_equipment', 'displayName': 'Planned Equipment Elements', 'media_types': ['csv', 'json', 'xls'] },
    { 'id': 3, 'reportType': 'GENERAL', 'name': 'planned_fibers', 'displayName': 'Planned Fiber Elements', 'media_types': ['csv', 'json', 'xls'] }
  ],
  reportBeingEdited: null
}

// -----------------------------------------------------------------------------
test('Default component render and click edit one report', () => {
  const mockGetReportsMetadata = jest.fn()
  const mockStartEditingReport = jest.fn()
  const component = shallow(
    <ReportDefinitionList reportsMetaData={reportsState.metaData} reportBeingEdited={null}
      getReportsMetadata={mockGetReportsMetadata} startEditingReport={mockStartEditingReport} />
  )
  expect(mockGetReportsMetadata.mock.calls.length).toBe(1)
  expect(component).toMatchSnapshot()

  component.find('#btnEditReport1').simulate('click')
  expect(mockStartEditingReport).toBeCalledWith(1)
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('Edit a report', () => {
  const mockGetReportsMetadata = jest.fn()
  const component = shallow(
    <ReportDefinitionList reportsMetaData={reportsState.metaData} reportBeingEdited={reportsState.metaData[1]}
      getReportsMetadata={mockGetReportsMetadata} />
  )
  expect(mockGetReportsMetadata.mock.calls.length).toBe(1)
  expect(component).toMatchSnapshot()
})
