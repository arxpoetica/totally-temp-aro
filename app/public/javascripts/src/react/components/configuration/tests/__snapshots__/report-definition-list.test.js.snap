// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default component render and click edit one report 1`] = `
<div
  id="divReportsList"
>
  <table
    className="table table-sm table-striped"
  >
    <tbody>
      <tr
        key="1"
      >
        <td>
          GENERAL
        </td>
        <td>
          plan_report
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport1"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          GENERAL
        </td>
        <td>
          Planned Equipment Elements
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport2"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        key="3"
      >
        <td>
          GENERAL
        </td>
        <td>
          Planned Fiber Elements
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport3"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Default component render and click edit one report 2`] = `
<div
  id="divReportsList"
>
  <table
    className="table table-sm table-striped"
  >
    <tbody>
      <tr
        key="1"
      >
        <td>
          GENERAL
        </td>
        <td>
          plan_report
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport1"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          GENERAL
        </td>
        <td>
          Planned Equipment Elements
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport2"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        key="3"
      >
        <td>
          GENERAL
        </td>
        <td>
          Planned Fiber Elements
        </td>
        <td>
          <button
            className="btn btn-primary"
            id="btnEditReport3"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Edit a report 1`] = `
<Connect(ReportDefinitionEditor)
  id="comReportDefinitionEditor"
/>
`;
